include/swift/Sema/ConstraintSystem.h

```diff
@@ -3976,9 +3976,13 @@ class ConstraintSystem {
   /// parent type by introducing fresh type variables for generic parameters
   /// and constructing a bound generic type from these type variables.
   ///
+  /// \param useArchetypes When \p decl has no parent, indicates whether to use
+  /// archetypes for generic parameters of its parent context.
```

Comment:
Even if there is a parent type, it might be nested inside of another non-type generic context, eg

```
func foo<T>() {
  struct Outer {
    struct Inner {
      typealias A = Array<T>
  }
}
```

Reply:
IIRC we don't support type declarations nested inside generic (or contextually generic) functions.

Reply:
I've changed the flag name to "isTypeResolution" to be more explicit.